src/hotspot/share/runtime/vframeArray.cpp

```diff
@@ -97,7 +95,15 @@ void vframeArrayElement::fill_in(compiledVFrame* vf, bool realloc_failures) {
         } else {
           assert(monitor->owner() == nullptr || !monitor->owner()->is_unlocked(), "object must be null or locked");
           dest->set_obj(monitor->owner());
+
+          assert(current_thread->is_Java_thread(), "Must be a JavaThread");
+          assert(ObjectSynchronizer::current_thread_holds_lock((JavaThread*) current_thread, Handle(current_thread, dest->obj())),
```

Comment:
This makes me wonder about the assert at line 96 that allows monitor->owner() == nullptr.  If that can happen due to OOM, then we need to check for that here too.

Reply:
OK yes - if monitor->owner() == nullptr
that's the MonitorInfo* we got from vframe->monitors() has a null Handle _owner

current_thread_holds_lock() is going to call read_stable_mark() on the oop, and likely crash, so this is a really good point!